ddtrace/contrib/asyncio/patch.py

```diff
@@ -15,6 +17,7 @@ def patch():
     if getattr(asyncio, "_datadog_patch", False):
         return
     setattr(asyncio, "_datadog_patch", True)
+    tracer.configure(context_provider=AsyncioContextProvider)
```

Comment:
Yeah I'm a bit uneasy about this. `asyncio` will be patched even if asyncio is not being used. So the `AsyncioContextProvider` will be installed and used for pretty much any Python 3 application.

Will think of whether there's a better entry point we can use to use. We face a similar problem in gevent where the GeventContextProvider is installed on the global tracer but if any custom tracer instances are instantiated they will not be configured correctly.

Reply:
yeah, I think the `tracer.configure()` is the part that I wish wasn't there, no real suggestions, but I get why we want to active it.

Reply:
write some auto-detection code in the tracer?
1. allow each integration to provide hints about the context provider that should be used
```
# in gevent integration
def advise_context_provider_to_use() -> ContextProvider:
    if gevent.monkey.patched:
       return GeventContextProvider
```
2. `Tracer` call all `advise_context_provider_to_use` for every integration that is enabled at setup time
3. if only one result is returned, it can use it; if multiple results are returned, e.g. gevent+asyncio, then just explain we can't do it (or make it work lol)

just an idea.

Reply:
Yes I like the idea of hooking into the tracer probably at the class-level so that all tracer instances can use the correct provider.

It's going to be tricky for asyncio as our patching will be applied before we can tell if asyncio is actually used... Maybe monkey patching `asyncio.run()` would be such an entry point 🙈